packages/nextjs/app/timeline/page.tsx

```diff
@@ -0,0 +1,136 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import { BatchTimeline } from "~~/components/BatchTimeline";
+import { useTimelineEvents } from "~~/hooks";
+
+const BATCH_REGISTRY_ADDRESS = "0x23E4943145668C06B55Bbc7cDEEEc6353687305B" as const;
+const BATCH_GRADUATION_NFT_ADDRESS = "0x23E4943145668C06B55Bbc7cDEEEc6353687305B" as const;
+const GITHUB_REPO_OWNER = "BuidlGuidl";
+const GITHUB_REPO_NAME = "batch21.buidlguidl.com";
+
+const LOADING_MESSAGES = [
+  "⛓️ Reading on-chain check-ins from Arbitrum",
+  "🎓 Fetching graduation NFT mints",
+  "✨ Loading merged PRs from GitHub",
+];
+
+export default function TimelinePage() {
```

Comment:
You might want to consider typing your page as NextPage since we are using TypeScript, as this ensures better integration and adds clarity for future maintainers by indicating that the component is a Next.js page.